skyportal/models.py

```diff
@@ -206,6 +208,71 @@ def desi_dr8_url(self):
         return (f"http://legacysurvey.org/viewer/jpeg-cutout?ra={self.ra}"
                 f"&dec={self.dec}&size=200&layer=dr8&pixscale=0.262&bands=grz")
 
+    def airmass(self, telescope, time, below_horizon=np.inf):
+        """Return the airmass of the Obj at time `time` from Telescope
+        `telescope`.
+
+        Parameters
+        ----------
+
+        telescope: skyportal.models.Telescope
+            The telescope to use for the airmass calculation
+
+        time: astropy.time.Time
+            The time or times at which to calculate the airmass
+
+        Returns
+        -------
+
+        secz: ndarray in the shape of `time`
+           The airmass of the Obj at the requested times
+        """
+
+        output_shape = np.shape(time)
+        time = np.atleast_1d(time)
+        altitude = self.altitude(telescope, time).to('degree').value
+
+        # use Pickering (2002) interpolation to calculate the airmass
+
+        with np.errstate(invalid='ignore'):
```

Comment:
Why this ignore?

Reply:
Because below the horizon, the altitude takes on a negative value, and numpy's power ufunc causes the negative values to be turned into `nan`s. I rewrote this chunk to avoid this behavior and not rely on `errstate`.